### PR TITLE
GH-1657: Intercom push notifications

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1657 

Changes proposed in this pull request:
- Register for Intercom Push notifications
- Display Intercom chat whn the user clicks on the push notification
- Make Intercom and Firebase work with Push notifications at the same time by disabling swizzling

### CODE REVIEW

File: Multisig/App/AppDelegate+Messaging.swift [hunk 1]
```diff
@@ -31,6 +32,11 @@ extension AppDelegate: UNUserNotificationCenterDelegate {
         let userInfo = response.notification.request.content.userInfo
 
         Messaging.messaging().appDidReceiveMessage(userInfo)
+        
+        if Intercom.isIntercomPushNotification(userInfo) {
+            LogService.shared.debug("PUSH: userNotificationCenter Intercom push notification with userInfo: \(userInfo)")
+            Intercom.presentMessenger()
```
Comment (sche) [inline Multisig/App/AppDelegate+Messaging.swift:38]: what does it do UI-wise?

Comment (biafra23) [reply Multisig/App/AppDelegate+Messaging.swift:38]: It opens the messenger as an overlay. As required in task two.

File: Multisig/App/AppDelegate.swift [hunk 2]
```diff
@@ -9,12 +9,15 @@
...
 import SwiftUI
 import Firebase
+import Intercom
 
 class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         App.shared.firebaseConfig.setUp()
         App.shared.intercomConfig.setUp()
 
+        UIApplication.shared.registerForRemoteNotifications()
```
Comment (biafra23) [inline Multisig/App/AppDelegate.swift:19]: Register on every app start. This does not trigger the messaging permission. The permission is till asked for when adding the first safe.

File: Multisig/App/AppDelegate.swift [hunk 3]
```diff
@@ -81,4 +84,23 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        LogService.shared.debug("PUSH: Received APNS token: \(deviceToken.toHexStringWithPrefix())")
+        Messaging.messaging().apnsToken = deviceToken
+        Intercom.setDeviceToken(deviceToken)
+    }
+
+    func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        LogService.shared.debug("PUSH: didReceiveRemoteNotification with userInfo: \(userInfo)")
+        Messaging.messaging().appDidReceiveMessage(userInfo)
+        
+        if Intercom.isIntercomPushNotification(userInfo) {
```
Comment (biafra23) [inline Multisig/App/AppDelegate.swift:98]: Not sure this is necessary.

File: Multisig/Info.plist [hunk 4]
```diff
@@ -205,5 +205,9 @@
 	<string>The app needs access to the photo library if you need to share your screenshot with our support team.</string>
 	<key>DESKTOP_PAIRING_EXPERIMENTAL_ENABLED</key>
 	<string>$(DESKTOP_PAIRING_EXPERIMENTAL_ENABLED)</string>
+	<key>FirebaseAppDelegateProxyEnabled</key>
```
Comment (biafra23) [inline Multisig/Info.plist:208]: Disable swizzling for Firebase and Intercom.

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**